src/absil/ilread.fs

```diff
@@ -1,3939 +1,2582 @@
-let readBlobHeapAsInt32 ctxt vidx = fst (sigptrGetInt32 (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsInt64 ctxt vidx = fst (sigptrGetInt64 (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsByte ctxt vidx = fst (sigptrGetByte (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsUInt16 ctxt vidx = fst (sigptrGetUInt16 (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsUInt32 ctxt vidx = fst (sigptrGetUInt32 (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsUInt64 ctxt vidx = fst (sigptrGetUInt64 (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsSingle ctxt vidx = fst (sigptrGetSingle (readBlobHeap ctxt vidx) 0) 
-let readBlobHeapAsDouble ctxt vidx = fst (sigptrGetDouble (readBlobHeap ctxt vidx) 0) 
-   
-//-----------------------------------------------------------------------
-// Some binaries have raw data embedded their text sections, e.g. mscorlib, for 
-// field inits. And there is no information that definitively tells us the extent of 
-// the text section that may be interesting data. But we certainly don't want to duplicate 
-// the entire text section as data! 
-//  
-// So, we assume: 
-//   1. no part of the metadata is double-used for raw data  
-//   2. the data bits are all the bits of the text section 
-//      that stretch from a Field or Resource RVA to one of 
-//        (a) the next Field or resource RVA 
-//        (b) a MethodRVA 
-//        (c) the start of the metadata 
-//        (d) the end of a section 
-//        (e) the start of the native resources attached to the binary if any
-// ----------------------------------------------------------------------*)
-
-// noFileOnDisk indicates that the PE file was read from Memory using OpenILModuleReaderFromBytes
-// For example the assembly came from a type provider
-// In this case we eagerly read the native resources into memory
-let readNativeResources (pectxt: PEReader) = 
-    [ if pectxt.nativeResourcesSize <> 0x0 && pectxt.nativeResourcesAddr <> 0x0 then 
-        let start = pectxt.anyV2P (pectxt.fileName + ": native resources", pectxt.nativeResourcesAddr)
-        if pectxt.noFileOnDisk then
-            let unlinkedResource =
-                let linkedResource = seekReadBytes (pectxt.pefile.GetView()) start pectxt.nativeResourcesSize
-                unlinkResource pectxt.nativeResourcesAddr linkedResource
-            yield ILNativeResource.Out unlinkedResource
-        else
-            yield ILNativeResource.In (pectxt.fileName, pectxt.nativeResourcesAddr, start, pectxt.nativeResourcesSize ) ]
-
-
-let getDataEndPointsDelayed (pectxt: PEReader) ctxtH = 
-    lazy
-        let (ctxt: ILMetadataReader) = getHole ctxtH
-        let mdv = ctxt.mdfile.GetView()
-        let dataStartPoints = 
-            let res = ref []
-            for i = 1 to ctxt.getNumRows TableNames.FieldRVA do
-                let rva, _fidx = seekReadFieldRVARow ctxt mdv i
-                res := ("field", rva) :: !res
-            for i = 1 to ctxt.getNumRows TableNames.ManifestResource do
-     
```

Comment:
Let's not read the entire block, should just use an instance of `ByteMemory` that spans over the specific content.